paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/CustomerSheetTest.kt

```diff
@@ -255,24 +342,42 @@ internal class CustomerSheetTest {
         context.markTestSucceeded()
     }
 
+    private fun billingDetails(): RequestMatcher {
+        return RequestMatchers.composite(
+            bodyPart(urlEncode("billing_details[address][postal_code]"), CustomerSheetPage.ZIP_CODE),
+            bodyPart(urlEncode("billing_details[address][country]"), CustomerSheetPage.COUNTRY)
+        )
+    }
+
+    private fun fullBillingDetails(): RequestMatcher {
+        return RequestMatchers.composite(
+            bodyPart(urlEncode("billing_details[name]"), urlEncode(CustomerSheetPage.NAME)),
+            bodyPart(urlEncode("billing_details[phone]"), CustomerSheetPage.PHONE_NUMBER),
+            bodyPart(urlEncode("billing_details[email]"), urlEncode(CustomerSheetPage.EMAIL)),
+            bodyPart(urlEncode("billing_details[address][line1]"), urlEncode(CustomerSheetPage.ADDRESS_LINE_ONE)),
+            bodyPart(urlEncode("billing_details[address][line2]"), urlEncode(CustomerSheetPage.ADDRESS_LINE_TWO)),
+            bodyPart(urlEncode("billing_details[address][city]"), urlEncode(CustomerSheetPage.CITY)),
+            bodyPart(urlEncode("billing_details[address][state]"), urlEncode(CustomerSheetPage.STATE)),
+            billingDetails()
+        )
+    }
+
     private fun cardDetails(
         cardNumber: String = CustomerSheetPage.CARD_NUMBER
     ): RequestMatcher {
         return RequestMatchers.composite(
             bodyPart("type", "card"),
-            bodyPart("card%5Bnumber%5D", cardNumber),
-            bodyPart("card%5Bexp_month%5D", CustomerSheetPage.EXPIRY_MONTH),
-            bodyPart("card%5Bexp_year%5D", CustomerSheetPage.EXPIRY_YEAR),
-            bodyPart("card%5Bcvc%5D", CustomerSheetPage.CVC),
-            bodyPart("billing_details%5Baddress%5D%5Bpostal_code%5D", CustomerSheetPage.ZIP_CODE),
-            bodyPart("billing_details%5Baddress%5D%5Bcountry%5D", CustomerSheetPage.COUNTRY)
+            bodyPart(urlEncode("card[number]"), cardNumber),
```

Comment:
So easy to read now! Thanks!

If you're up for it, might be good to have a follow up for the other areas that do this as well!